Migration rollbacks for event databases

Molrn opened the pull request:
add the command `python utils/event_databases/migrate.py`:
````sh
Command migrating one or more event databases to a specific version.

options:
  -h, --help            show this help message and exit
  -v VERSION, --version VERSION
                        Version of the database to migrate to. Defaults to the last version.
  -e EVENT, --event EVENT
                        ID of the event to migrate. Defaults to all the events.
````  

Migrating to a previous version uses the backward methods defined in the EventMigration classes. By default that method raises a `NotImplementedError`. This disables the rollback feature for migrations we don't want / can't implement the backward method of (ex: 2.4.21)

Molrn committed:
Migration rollbacks for event databases

Molrn committed:
fix command description

pascalaubry commented on utils/event_databases/restore.py:
```diff
@@ -42,38 +40,38 @@
         version = Version(args.version)
         if version > PapiWebConfig.version:
-            logger.error(
+            sys.stderr.write(
```
Can you use print_interactive_error() and print_interactive_info() instead?

pascalaubry commented on utils/event_databases/migrate.py:
```diff
@@ -0,0 +1,51 @@
+    for event_id in event_ids:
+        with EventDatabase(event_id, True, False) as event_database:
+            if event_database.version == version:
+                sys.stdout.write(
```
Can you use print_interactive_info() instead?

pascalaubry commented on utils/event_databases/migrate.py:
```diff
@@ -0,0 +1,51 @@
+            'Defaults to the last version.'
+        ),
+    )
+    parser.add_argument(
```
Can you add a --all-events options to apply to all of the databases and make one of --all-events or --event mandatory?
(I was really surprised to see the script applying to all my databases by default...)

pascalaubry committed:
Merge branch 'dev' into migration-rollback

pascalaubry commented on src/database/sqlite/event_migration.py:
```diff
@@ -46,49 +59,98 @@ def last_migration_version(self) -> Version:
+                    f'to version {migration_version}: "{e}"'
+                )
+
+    def _rollback(
```
![image](https://github.com/user-attachments/assets/1432dfab-12db-4d1a-a6fb-c63f745bd368)
Instead of throwing exceptions in this method, could you print a simple error message and return a boolean?

pascalaubry reviewed (changes requested):
Apart from a few comments, works.
Still thinking that it is only comfort for developers, but OK let's cherish ourselves :-)

Molrn committed:
fixes

- use interactive prints
- --all-events option
- remove error raised in migrate function

Molrn commented:
for now it is comfort for devs, but it's the kind of stuff that might come in handy if one day the app is deployed and something needs to be fixed in production ;) and comfort for devs is rather important for OS projects :)

pascalaubry reviewed (approved):
Thanks.
